packages/api-gateway-service/package.json

```
@@ -37,9 +37,10 @@
   "dependencies": {
     "@apollo/gateway": "0.16.6",
     "@types/jsonwebtoken": "8.5.0",
-    "@types/ws": "7.2.4",
-    "apollo-link-http": "1.5.17",
-    "apollo-link-ws": "1.0.20",
+    "@types/ws": "^7.2.4",
+    "apollo-link-http": "^1.5.17",
+    "apollo-link-retry": "^2.2.16",
+    "apollo-link-ws": "^1.0.20",
```

```suggestion
    "@types/ws": "7.2.4",
    "apollo-link-http": "1.5.17",
    "apollo-link-retry": "2.2.16",
    "apollo-link-ws": "1.0.20",
```
Pin these dependencies!

---

Updated